torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1066,7 +1084,9 @@ void ProcessGroupNCCL::dumpDebuggingInfo() {
       registerDebugInfoWriter(std::move(debugInfoWriterPtr));
     }
     debugInfoWriter_->write(ncclTrace);
+    return true;
```

Comment:
annoyingly it looks like you need a real value for promise.setValue, so i went with a bool and it has some limited utility in showing which side of this if we landed on.

Reply:
I mean do we care about the return value or not?

Reply:
i didn't bother checking it. i supposed its cleaner to actually check that it's true.